Sources/CryptoBoringWrapper/EC/EllipticCurvePoint.swift

```diff
@@ -12,41 +12,262 @@
+    }
+
+    @usableFromInline
+    package func subtract(_ rhs: EllipticCurvePoint, on group: BoringSSLEllipticCurveGroup) throws {
+        try self.add(rhs.inverting(on: group), on: group)
+    }
+
+    @usableFromInline
+    package convenience init(subtracting rhs: EllipticCurvePoint, from lhs: EllipticCurvePoint, on group: BoringSSLEllipticCurveGroup) throws {
+        try self.init(copying: lhs, on: group)
+        try self.subtract(rhs, on: group)
+    }
+
+    @usableFromInline
+    package func subtracting(_ rhs: EllipticCurvePoint, on group: BoringSSLEllipticCurveGroup) throws -> EllipticCurvePoint {
+        try EllipticCurvePoint(subtracting: rhs, from: self, on: group)
+    }
+
+    @usableFromInline
+    package static func subtracting(_ rhs: EllipticCurvePoint, from lhs: EllipticCurvePoint, on group: BoringSSLEllipticCurveGroup) throws -> EllipticCurvePoint {
+        try EllipticCurvePoint(subtracting: rhs, from: lhs, on: group)
+    }
+
+    @usableFromInline
+    package init<MessageBytes: ContiguousBytes, DSTBytes: ContiguousBytes>(hashing msg: MessageBytes, to group: BoringSSLEllipticCurveGroup, domainSeparationTag: DSTBytes) throws {
+        let hashToCurveFunction = switch group.curveName {
+        case .p256: CCryptoBoringSSLShims_EC_hash_to_curve_p256_xmd_sha256_sswu
+        case .p384: CCryptoBoringSSLShims_EC_hash_to_curve_p384_xmd_sha384_sswu
+        case .p521: throw CryptoBoringWrapperError.invalidParameter // BoringSSL doesn't have a hash_to_curve API for P521.
+        case .none: throw CryptoBoringWrapperError.internalBoringSSLError()
+        }
+
+       self._basePoint = try msg.withUnsafeBytes { msgPtr in
+            try group.withUnsafeGroupPointer { groupPtr in
+                try domainSeparationTag.withUnsafeBytes { dstPtr in
+                    guard
+                        let pointPtr = CCryptoBoringSSL_EC_POINT_new(groupPtr),
+                        hashToCurveFunction(
+                            groupPtr,
+                            pointPtr,
+                            dstPtr.baseAddress,
+                            dstPtr.count,
+                            msgPtr.baseAddress,
+                            msgPtr.count
+                        ) == 1
+                    else { throw CryptoBoringWrapperError.internalBoringSSLError() }
+                    return pointPtr
+                }
+            }
+        }
+    }
+
+    @usableFromInline
+    package func isEqual(to rhs: EllipticCurvePoint, on group: BoringSSLEllipticCurveGroup) -> Bool {
+        self.withPointPointer { selfPtr in
+            group.withUnsafeGroupPointer { groupPtr in
+                rhs.withPointPointer { rhsPtr in
+                    switch CCryptoBoringSSL_EC_POINT_cmp(groupPtr, selfPtr, rhsPtr, nil) {
+                    case 0: return true
+                    case 1: return false
+                    default: fatalError("Unexpected error testing EC point equality")
```

Comment:
I don't know that we need to be this strict. `0` is true, everything else is `false`. Happy to have an assertion for the other codes, but probably shouldn't fatal error.

Reply:
I think that's not quite right since the docs say that `EC_POINT_cmp` can return an error: 

```
EC_POINT_cmp returns 1 if the points are not equal, 0 if they are, or -1 on error.
```

I think this can happen if you provide a point on a different group for an equality test, for instance.

This function isn't throwing so the best we can do is fatal error, which seems to be in line with what we do elsewhere?

Reply:
After further consideration, given we allow folks to pass in the `group`, we shouldn't even assert, but just return `false` if comparing two points from different groups.

I've updated the code to clear the BoringSSL error and return `false` in this scenario.